### DISCUSSION TITLE
How to debug default updater

### DISCUSSION THREAD

**Discussion by mvdschee** [tauri-apps/tauri#5282] [Q&A] [answered]
I'm running in to a problem with the default updater, so it show me correctly that there is an update and i do see it's downloading a large file on my network but after the download stops it shows me nothing (also I see no popup for download progress not sure if that is supposed to happen or not). 

I have seen you can implement your own updater but I rather stick with the provided one, but how do you debug or at least have some errors popup?

Thank you in advanced!

**Comment by FabianLars** [comment, marked as answer]
If Rust is an option, you could print the RunEvent::Updater events in the [`run`](https://docs.rs/tauri/latest/tauri/struct.App.html#method.run) method.

Otherwise you can listen to the `"tauri://update-status"` event in javascript, but you need to trigger the update manually after that by emitting the `"tauri://update"` event _after registering the listener_, because the default auto updater runs too early.

**Reply by mvdschee** [reply]
That works great for debugging and found my error (Invalid signature 🤔) 

Right now I have this code that is loaded on startup but from what I understand from the docs this should not work, but I guess were are triggering the update now twice and that should be good for debugging. 
```ts
import { listen, emit } from '@tauri-apps/api/event';

listen('tauri://update-status', function (res) {
    console.log(res);
});

emit('tauri://update');
```

**Reply by FabianLars** [reply]
> but I guess were are triggering the update now twice

Yep!
